crates/topos-tce-transport/src/lib.rs

```diff
@@ -101,14 +114,80 @@ pub enum ProtocolEvents {
     /// Indicates that 'echo' message broadcasting is required
     Echo {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// Indicates that 'ready' message broadcasting is required
     Ready {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// For simulation purpose, for now only caused by ill-formed sampling
     Die,
 
     /// Stable Sample
     StableSample,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum Error {}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct ValidatorId(H160);
+
+impl ValidatorId {
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    pub fn address(&self) -> Address {
+        Address::from_str(&format!("0x{}", hex::encode(self.0))).unwrap()
+    }
+}
+
+impl From<H160> for ValidatorId {
+    fn from(address: H160) -> Self {
+        ValidatorId(address)
+    }
+}
+
+impl From<&str> for ValidatorId {
+    fn from(address: &str) -> Self {
+        ValidatorId(H160::from_str(address).expect("Cannot parse address string as H160"))
+    }
+}
+
+impl std::fmt::Display for ValidatorId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "0x{}", hex::encode(self.0))
+    }
+}
+
+pub async fn sign_message(
+    validator_id: ValidatorId,
+    certificate_id: CertificateId,
+    wallet: LocalWallet,
```

Comment:
You don't need ownership to sign here
```suggestion
    wallet: &LocalWallet,
```